### ISSUE TITLE
PHP-based search

### ISSUE THREAD

**Issue by EzequielBruni** [typemill/typemill#330]
I don't feel entirely comfortable leaving a navigational function as important as search in the hands of JavaScript. It's the kind of thing that can just... break.

A PHP-based search plugin and the ability to customize a search results page would be much appreciated. Again, I know this is another big ask... but as long as we're all just making feature requests... you know?

**Comment by trendschau** [comment]
Yes, I thought the same originally, but when I created the plugin I searched for a php-search-library and there is no such library that

* is lightweight
* does not care if files ore database
* is supported and regularly updated 
* is widespread

Lunr is all of this so for me it is the only choice for a project like typemill. Maybe someone has other experiences and knows a lightweight php library for it.

**Comment by trendschau** [comment]
Excited about the improvements, if useful for all themes I will migrate to the plugin. Another idea was to open the results in a separate page, but the problem then is the styling according to the theme, so there is no really super good solution for it all yet ...

**Comment by EzequielBruni** [comment]
Oh, I wouldn't worry about me making "improvements". I mean, I did make a couple of accessibility changes to the HTML form, but the biggest changes came from me removing all the default styling and moving the actual CSS implementation to my theme so I could make it look exactly how I want.

While that might prove useful to some people, it's not exactly an improvement.
